modules/gapi/src/compiler/gislandmodel.hpp

```diff
@@ -112,6 +114,9 @@ class GIslandExecutable
     virtual bool canReshape() const = 0;
     virtual void reshape(ade::Graph& g, const GCompileArgs& args) = 0;
 
+    virtual void compile(const std::vector<Data>&,
+                         const std::vector<Data>&) {};
```

Comment:
Instead of this, I'd extend introduce a new `GBackend::Priv`'s `compile()` with this new parameters. The old version could be called in its default implementation by default - for compatibility with the existing Backends. If a backend is _capable_, it would override this new `compile()`, old versions would work for the rest.

Your PlaidML backend would override this new version and pass that parameter to PlaidMLExecutable's constructor explicitly. AFIAR, executable contrustors are free-form.

Reply:
Fixed